runtime/recipe-index.js

```diff
@@ -99,7 +100,39 @@ export class RecipeIndex {
         results.push(otherHandle);
       }
     }
-
     return results;
   }
+
+  // Given a slot, find consume slot connections that could be connected to it.
+  findConsumeSlotConnectionMatch(slot) {
+    let consumeConns = [];
+    for (let recipe of this._recipes) {
+      for (let slotConn of recipe.slotConnections) {
+        if (!slotConn.targetSlot && MapSlots.specMatch(slotConn, slot) && MapSlots.tagsOrNameMatch(slotConn, slot)) {
+          let matchingHandles = [];
+          if (!MapSlots.handlesMatch(slotConn, slot)) {
+            // Find potential handle connections to coalesce
+            slot.handleConnections.forEach(slotHandleConn => {
+              let matchingConn = Object.values(slotConn.particle.connections).find(particleConn => Handle.effectiveType(slotHandleConn.handle._mappedType, [particleConn]));
```

Comment:
Should we produce one result per matchingConn instead of taking only the first one? (i.e. find => filter ?)

Reply:
done